EventFilter/L1TRawToDigi/plugins/implementations_stage2/BMTFPackerInputs.cc

```diff
@@ -0,0 +1,367 @@
+#include "BMTFPackerInputs.h"
+
+#include <vector>
+//#include <bitset>//debug
+
+
+namespace l1t 
+{
+   namespace stage2 
+   {
+
+      const int BMTFPackerInputs::ownLinks_[]={4,5,12,13,20,21,22,23,28,29};
+
+      Blocks BMTFPackerInputs::pack(const edm::Event& event, const PackerTokens* toks)
+      {
+	int board_id = (int)board();
+
+        auto muonPhToken = static_cast<const BMTFTokens*>(toks)->getInputMuonTokenPh();
+        auto muonThToken = static_cast<const BMTFTokens*>(toks)->getInputMuonTokenTh();
+
+        Blocks blocks;
+  
+        edm::Handle<L1MuDTChambPhContainer> phInputs;
+        event.getByToken(muonPhToken, phInputs);
+        edm::Handle<L1MuDTChambThContainer> thInputs;
+        event.getByToken(muonThToken, thInputs);
+
+	uint32_t qualEta_32bit = 0;
+	uint32_t posEta_0_32bit = 0;	
+	uint32_t posEta_n2_32bit = 0, posEta_n1_32bit = 0;
+	uint32_t posEta_p2_32bit = 0, posEta_p1_32bit = 0;
+
+	bool moreBXeta = false;
+	for (int link = 0; link <= 35; link++) {
+	  //	  std::cout << "link : " << link << std::endl;
+
+	  if ( (link >= 6 && link < 8) ||
+	       (link >= 14 && link < 16) ||
+	       (link >= 30 && link < 32) ) 
+	    continue;
+
+	  //initializing null block_payloads and block_id
+	  std::vector<uint32_t> payload_0(6,(uint32_t)0);
```

Comment:
the (uint_32_t) are not needed

Reply:
it looks like lots of code gets simpler if these payloads are declared as
std::array< std::vector<uint32_t>,5>
or even
std::vector<uint32_t> payloads(5*6,0)

Reply:
(plus an enum for readability where a simple loop is not appropriate)